### DISCUSSION TITLE
Sync-lock Off Warning + Favorites

### DISCUSSION THREAD

**Discussion by TheWendyPower** [tenacityteam/tenacity-legacy#388] [Ideas]
One of the biggest frustrations that I have when editing multiple voice tracks, is forgetting to turn sync-lock back on. I will unlock tracks to move something around and forget to turn it back on before I continue to edit a single voice track, Then the conversation becomes out of sync and I have to mess with the other tracks to get the timing back on. Because not everyone needs/would find this feature useful, placing the ability to turn on/off the warning in settings would also be useful. 

Another thing would be a favorite menu for effects. Of the many that can be used 6 are all, I need for my projects. Being able to have a shortlist of those effects would shave a bit of time off the editing process. 

Thank you for all the work you have put into this project so far. I'm really enjoying it and look forward to seeing it grow in the future.

**Comment by akleja** [comment]
Instead of a warning, perhaps there's another way to tackle the issue. What if when sync-lock is on, holding say [ctrl] would allow you to temporarily override the sync-lock?

I agree that a favorite menu would be nice for the effects. In the meantime you can just disable the effects you never use by going to Add/Remove Plug-ins in the effects menu.
I also think it would be a huge improvement if the effects could be sorted into categories (Eq/Dynamics/Modulation/etc.)

**Reply by poyribn** [reply]
Yes. Sorting the effects into categories will be really great.

---

I'd also propose visualizing the audio effects with icons, similar to how New&nbsp;Inkscape visualizes their path effects.
![](https://wiki.inkscape.org/wiki/images/5/5e/Add_LPE_dialog.png)

Of course, it'll be harder to come up with visual representations of audio effects than it is to come up with visual representations of visual effects,
since we'd have to represent one medium (audio) through another medium (visuals).

But if the icons communicate the audio concepts well, it'll make picking the right audio effects much faster and easier.

**Reply by TheWendyPower** [reply]
A temporary override of sync-lock may be a very useful way to quickly move small sections, while minimizing the risk of messing up the flow of the other tracks.

**Reply by akleja** [reply]
@TheWendyPower I've made a PR where the alt key to temporary disables the lock, if you want to try it out: https://github.com/tenacityteam/tenacity/pull/404

**Comment by akleja** [comment]
Yeah I really like the idea with icons similar to inkscape.
One problem might be when using external effects (lv2, laadspa, vst etc). Most of them do ship with icons, but it will probably not look coherent, and the icons are usually more of a branding thing.

**Reply by poyribn** [reply]
We first need the style of the built-in icons locked in, which would take a while.

After that we could make our own icons (probably with loss of branding if it makes the icon more descriptive) for some of the most popular plug-ins.

And after That we could write a sptyleguide for Tenacity plug-in icons. It'll probably include something along the lines of "As these icons are meant to be purely descriptive, you might have to remove the branding or parts of the branding to make the icon more descriptive".

This problem of incosistency will never be fully solved (as there will always be obscure old plug-ins from before Tenacity released), but it will be much more solved than before.

**Reply by akleja** [reply]
I think perhaps sticking with custom drawn icons for the built in effects and for effect categories might be a good idea.
Having icons for some external plugins, and not others seem more confusing than helping.

I mean I have a few hundreds plugins. While I don't use all of them every day, I still use most of them occasionally. Those I'd prefer in a list. What matters most to me is ease in finding the right one quickly.
That would mean a good search function and clear customizable categories.
